core/src/main/scala/kafka/admin/ConfigCommand.scala

```diff
@@ -251,6 +253,23 @@ object ConfigCommand extends Config {
           s"This configuration will be ignored if the version is newer than the inter.broker.protocol.version specified in the broker.")
       }
     }
+    resolveVariableConfigs(opts, props)
+  }
+
+  private[admin] def resolveVariableConfigs(opts: ConfigCommandOptions, propsOriginal: Properties): Properties = {
```

Comment:
According to KIP-421, we were planning to support dynamic configs with config providers where the configs are resolved by the broker. With this approach, broker never sees the externalized confg. The configs stored by the broker will be the resolved values (e.g. unencrypted values that rely solely on the existing encryption in ZK). What value does external configs add to broker?

Reply:
You are right the resolution should not happen in ConfigCommand but on the broker side. On further debugging the issue, it is actually doing the auto resolution on the broker side. The issue is that it validates the config values before processing it. And as we are passing a string instead of int it fails the validation.